plugins/client/graphql-kotlin-client-generator/src/main/kotlin/com/expediagroup/graphql/plugin/client/generateClient.kt

```diff
@@ -42,7 +47,19 @@ fun generateClient(
         customScalarMap = customScalars,
         serializer = serializer
     )
-    val graphQLSchema = SchemaParser().parse(schema)
+    val graphQLSchema = SchemaParser().parse(getSchemaReader(schema))
     val generator = GraphQLClientGenerator(graphQLSchema, config)
     return generator.generate(queries)
 }
+
+private fun getSchemaReader(schemaPath: String): Reader =
+    if (schemaPath.startsWith("classpath://")) {
+        val resource = object {}.javaClass.getResource(URI(schemaPath).path) ?: throw RuntimeException("specified GraphQL schema not found, $schemaPath")
```

Comment:
i think anonymous objects like this might be throwing some detekt violations -> instead lets move the logic of reading from classpath to the `GraphQLClientGenerator` (i.e. change its constructor from accepting type registry to a path and initialize type registry in init block)

```
private val graphQLSchema: TypeDefinitionRegistry
init {
    graphQLSchema = parseSchema(schemaPath)
}

private fun parseSchema(path: String): TypeDefinitionRegistry {
    val schemaFile = File(path)
    return if (schemaFile.isFile) {
        SchemaParser().parse(schemaFile)
    } else {
        val schemaInputStream = this.javaClass.classLoader.getResourceAsStream(path) ?: throw RuntimeException("specified schema file=$path does not exist")
       SchemaParser().parse(schemaInputStream)
    }
}
```

we'll need some unit tests around this